test/test_v2/test_common.c

```diff
@@ -381,12 +392,174 @@ void pmixt_post_finalize(pmix_proc_t *this_proc, test_params *params, validation
+                    key, value->type));
+                    exit(1);
+                }
+                if (PMIXT_CHECK_KEY(key, PMIX_NSPACE) && val_params->check_pmix_nspace) {
+                    if (0 != strcmp(val_params->pmix_nspace, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_nspace));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_JOBID) && val_params->check_pmix_jobid) {
+                    if (0 != strcmp(val_params->pmix_jobid, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_jobid));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_LOCAL_PEERS) && val_params->check_pmix_local_peers) {
+                    if (0 != strcmp(val_params->pmix_local_peers, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_local_peers));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                }
+                else if (PMIXT_CHECK_KEY(key, PMIX_HOSTNAME) && val_params->check_pmix_hostname) {
+                    if (0 != strcmp(val_params->pmix_hostname, stringdata)) {
+                        TEST_ERROR(("Key %s failed validation. Get value: %s Validation value: %s",
+                                     key, stringdata, val_params->pmix_hostname));
+                        exit(1);
+                    }
+                    TEST_VERBOSE(("Namespace %s: Rank %d Data validated: Key: %s Value: %s",
+                        myproc->nspace, myproc->rank, key, stringdata));
+                } // other possibilities will be added here later
+                else {
+                    TEST_VERBOSE(("Check input for case PMIX_STRING: key: %s stringdata: %s",
+                        key, stringdata));
+                    exit(1);
+                }
+                free(stringdata);
+                break;
+            default:
+                TEST_VERBOSE(("No test logic for type: %d, key: %s", value->type, key));
```

Comment:
You might consider making the `default` case an error case. This would make it easier to see if one of the cases missed.

Reply:
Yes, sounds good.

Reply:
Should be fixed.